### DISCUSSION TITLE
Color/texture embeded in an .obj file

### DISCUSSION THREAD

**Discussion by Joaopmoliveira** [vsg-dev/VulkanSceneGraph#840] [Q&A] [answered]
Good afternoon vsg community, I was loading obj files into my custom viewer, but the color associated with the obj file is not rendered. Is this a default that I can change by tweaking the vsg::Options instance or it is impossible to extract this information from the vsg::read call because that is the desired behavior? 
I tried to look at the examples but I could not find anything too obvious at a first glance that solves my problem. 
Thank you for any feedback

**Comment by vsg-dev** [comment, marked as answer]
.obj files are currently loaded by the vsgXchange:assimp loader . You can list the available options by using vsgconv --features i.e.

`$ vsgconv --features vsgXchange::assimp`

The relevant part is:

```
   vsg::Options::Value      type
    -------------------      ----
    crease_angle             float
    discard_empty_nodes      bool
    generate_sharp_normals   bool
    generate_smooth_normals  bool
    two_sided                bool
```

There isn't any option listed for colours so I expect what you are seeing is simply a mapping of what vsgXchange::assimp is doing with what assimp provides from loading the .obj.  The discrepancy could be due to vsgXchange::assimp or assimp, or perhaps just an oddity in the .obj file you have.

As general note, .obj is a very very very old format, is really isn't a good interchange format, so I'd recommend seeing if you can load your data via a more modern and capable format such as .gltf.
